tests/integration/test_snapshot_option_extension_pythonpath.py

```diff
@@ -0,0 +1,84 @@
+from pathlib import Path
+
+import pytest
+
+
+@pytest.fixture
+def testfile(testdir):
+    testdir.makepyfile(
+        __init__="",
+        extension_file=(
+            """
+            import syrupy
+            class MySingleFileExtension(
+                syrupy.extensions.single_file.SingleFileSnapshotExtension
+            ):
+                pass
+            """
+        ),
+        test_file=(
+            """
+            def test_default(snapshot):
+                assert b"default extension serializer" == snapshot
+            """
+        ),
+    )
+    return testdir
+
+
+def test_snapshot_default_extension_option_success(testfile):
+    testfile.makeini(
+        f"""
+        [pytest]
+        pythonpath =
+            {testfile.tmpdir}
+    """
+    )
+
+    result = testfile.runpytest(
+        "-v",
+        "--snapshot-update",
+        "--snapshot-default-extension",
+        "extension_file.MySingleFileExtension",
+    )
+    result.stdout.re_match_lines((r"1 snapshot generated\."))
+    assert Path(
+        testfile.tmpdir, "__snapshots__", "test_file", "test_default.raw"
+    ).exists()
+    assert not result.ret
+
+
+def test_snapshot_default_extension_option_module_not_found(testfile):
+    result = testfile.runpytest(
+        "-v",
+        "--snapshot-update",
+        "--snapshot-default-extension",
+        "extension_file.MySingleFileExtensions",
```

Comment:
Hmm, this is a typo. The snapshot is successfully generated if I replace this with `extension_file.MySingleFileExtension`, even though I haven't customized `pythonpath` in this test case. This could mean project has been honoring pytest's import paths all along. I think this PR still fixes a bug, but I'm not sure the description is right. I converted this PR to draft. Let me investigate.

Reply:
I figured it out. A few lines above [this comment from #667](https://github.com/tophat/syrupy/pull/667/files#r1059134028), there's an in-memory cache. I didn't quite grok why the cache is there, but assuming it is necessary, the test module in this PR now clears the cache between test cases. I fixed the typo on this line. Tests pass as expected, locally.